libs/datamodel/connectors/mongodb-datamodel-connector/src/lib.rs

```diff
@@ -75,6 +75,29 @@ impl MongoDbDatamodelConnector {
             span: field.ast_field().span,
         });
     }
+
+    fn validate_array_native_type(field: ScalarFieldWalker<'_, '_>, errors: &mut datamodel_connector::Diagnostics) {
+        let (ds_name, type_name, args, span) = match field.raw_native_type() {
+            Some(nt) => nt,
+            None => return,
+        };
+
+        if type_name == type_names::ARRAY {
```

Comment:
I'd like this to avoid unnecessary nesting as much as possible. This could be:

```rust
if type_name != type_names::ARRAY {
    return;
}
```

Then you could shift the rest of the function one tab to the left and it looks much better.